.github/workflows/ci.yml

```diff
@@ -0,0 +1,24 @@
+name: CI
+on:
+  pull_request:
+    branches: [ "main" ]
+    
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+      - name: Setup Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: '1.22.2'
+
+      - name: Install dependencies
+        run: go get .
+        
+      - name: Test with the Go 
+        run: make test
+
+      - name: Build
+        run: make build 
```

Comment:
```suggestion

```
We can probably remove this build step for now actually. Running the tests is probably sufficient here.
Eventually, we could have some automation for the build step, but it probably makes sense for that to be a separate future workflow.